tests/util/rest.go

```diff
@@ -52,3 +69,72 @@ func FormFromFile(t *testing.T, form map[string]string) (*bytes.Buffer, string)
 
 	return body, mp.FormDataContentType()
 }
+
+func RetrievePipelineId(t *testing.T, APIServerURL string, PipelineDisplayName string) string {
+	response, err := http.Get(fmt.Sprintf("%s/apis/v2beta1/pipelines", APIServerURL))
+	require.NoError(t, err)
+	responseData, err := io.ReadAll(response.Body)
+	require.NoError(t, err)
+	var pipelineData Pipeline
+	var pipelineID string
+	err = json.Unmarshal(responseData, &pipelineData)
+	require.NoError(t, err)
+	for _, pipeline := range pipelineData.Pipelines {
+		if pipeline.DisplayName == PipelineDisplayName {
+			pipelineID = pipeline.PipelineID
+			break
+		}
+	}
+	return pipelineID
+}
+
+func FormatRequestBody(t *testing.T, pipelineID string, PipelineDisplayName string) []byte {
+	requestBody := PipelineRequest{
+		DisplayName: PipelineDisplayName,
+		PipelineVersionReference: struct {
+			PipelineID string `json:"pipeline_id"`
+		}{PipelineID: pipelineID},
+	}
+
+	body, err := json.Marshal(requestBody)
+	require.NoError(t, err)
+	return body
+}
+
+func WaitForPipelineRunCompletion(t *testing.T, APIServerURL string) error {
+	timeout := time.After(6 * time.Minute)
+	ticker := time.NewTicker(60 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("timed out waiting for pipeline run completion")
+		case <-ticker.C:
+			// Check the status of the pipeline run
+			status, err := CheckPipelineRunStatus(t, APIServerURL)
+			require.NoError(t, err)
+			if status == "SUCCEEDED" {
+				return nil
+			}
```

Comment:
what about the other statuses? if it's not SUCCEEDED, but it returned successfully, with a failure state for instance, we should return, instead of waiting for a full time out